src/JUDI.jl

```diff
@@ -15,6 +15,7 @@ using Distributed
 using DSP, FFTW, Dierckx
 using PyCall
 using JOLI, SegyIO
+using ChainRulesCore
```

Comment:
Sorry didn't follow. Why don't we do

```julia
using SpecialFunctions.ChainRulesCore
import SpecialFunctions.ChainRulesCore.rrule
```

like in JOLI?

Reply:
Sorry wrong comments.

JOLI does it in an ugly way to avoid extra dependency to keep it what it is. But this is much cleaner.

Reply:
But this makes ChainRulesCore as dependency

Reply:
Yes that's the point, if a pakcage depend on another one it should be a dependency

Reply:
"It would be done through require so it JUDI would stay independent of Chains rules but if a package is existing in the user Julia env it would load the rules"

Is there anyway to do it via require?